src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -420,26 +420,29 @@ public void Done()
 
             public bool SupportedDiagnosticId(ProjectId projectId, string id)
             {
-                if (_diagnosticIdMap.TryGetValue(projectId, out var ids))
+                lock (_diagnosticIdMap)
                 {
-                    return ids.Contains(id);
-                }
+                    if (_diagnosticIdMap.TryGetValue(projectId, out var ids))
```

Comment:
I dont think I can use MultiDictionary. I started to add new functionality to MultiDictionary and I end up use only ones I added and not existing functionality. actually existing behavior works against my need.